pkg/apis/extensions/register.go

```diff
@@ -73,24 +74,22 @@ func addKnownTypes(scheme *runtime.Scheme) {
 	)
 }
 
-func (obj *ClusterAutoscaler) GetObjectKind() unversioned.ObjectKind           { return &obj.TypeMeta }
-func (obj *ClusterAutoscalerList) GetObjectKind() unversioned.ObjectKind       { return &obj.TypeMeta }
-func (obj *Deployment) GetObjectKind() unversioned.ObjectKind                  { return &obj.TypeMeta }
-func (obj *DeploymentList) GetObjectKind() unversioned.ObjectKind              { return &obj.TypeMeta }
-func (obj *DeploymentRollback) GetObjectKind() unversioned.ObjectKind          { return &obj.TypeMeta }
-func (obj *HorizontalPodAutoscaler) GetObjectKind() unversioned.ObjectKind     { return &obj.TypeMeta }
```

Comment:
Yes, people are currently using them. I haven't removed `HorizontalPodAutoscaler` and `HorizontalPodAutoscalerList` types from `extensions/v1beta1`. I moved internal representation from `extensions/types.go` to `autoscaling/types.go` and updated `extensions/register.go` to point to `autoscaling.HorizontalPodAutoscaler` and `&autoscaling.HorizontalPodAutoscalerList` instead.

So AFAIU currently with support for both `autoscaling/v1` and `extensions/v1beta1` users will be able to perform the migration.